### PR TITLE
Fixes XML documentation for libraries.

### PR DESCRIPTION
**Pull request by gabog**
Addresses: #2076
Updated all library projects to generate Xml documentaiton for the release target.
Removed unused Documentaion build configuration.
Removed junk xml files.

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll**

File: libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj [hunk 1]
```diff
@@ -4,14 +4,19 @@
...
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <SignAssembly>true</SignAssembly>
     <DelaySign>true</DelaySign>
     <AssemblyOriginatorKeyFile>..\..\build\35MSSharedLib1024.snk</AssemblyOriginatorKeyFile>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <DocumentationFile>bin\$(Configuration)\netstandard2.0\Microsoft.Bot.Configuration.xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DocumentationFile>bin\$(Configuration)\netstandard2.0\Microsoft.Bot.Configuration.xml</DocumentationFile>
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj]: I don't think we want this one - no need to generate the XML File for the Debug build.

Comment (gabog) [reply libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj]: Sorry @cleemullins , didn't see the comment, will address in a new PR.

Comment (gabog) [reply libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj]: Addressed in #2085 2085

**Pull request merged**